streams/shared/src/main/scala/zio/stream/experimental/ZStream.scala

```diff
@@ -213,22 +267,21 @@ class ZStream[-R, +E, -M, +B, +A](
     } yield b
 
   /**
-   * Runs the stream and collects all of its elements in a list.
-   *
-   * Equivalent to `run(Sink.collectAll[A])`.
-   *
-   * @return an action that yields the list of elements in the stream
-   */
-  final def runCollect: ZIO[R, E, List[A]] = runQuery(ZSink.collectAll[A])
-
-  /**
-   * Runs the stream and collects all of its elements in a list.
-   *
-   * Equivalent to `run(Sink.collectAll[A])`.
-   *
-   * @return an action that yields the list of elements in the stream
+   * Takes the specified number of elements from this stream.
+   * @param n the number of elements to retain
+   * @return a stream with `n` or less elements
    */
-  final def runCollectManaged: ZManaged[R, E, List[A]] = runQueryManaged(ZSink.collectAll[A])
+  def take(n: Long): ZStream[R, E, M, Option[B], A] =
+    ZStream {
+      for {
+        as      <- self.process
+        counter <- Ref.make(0L).toManaged_
+        pull = counter.get.flatMap { c =>
+          if (c >= n) Pull.end(None)
```

Comment:
The question is do we want the marker to be permanently `Some(b)` in the event the stream is smaller than n elements and downstream keep pulling ad infinitum, or is it fine to eventually switch to `None`?

Reply:
Ha scratch that, the update will only happen if the pull is successful 🤦‍♂